substratest/factory.py

```diff
@@ -179,6 +179,7 @@ class DatasetSpec(_Spec):
     data_opener: str
     type: str
     description: str
+    metadata: typing.Dict = None
```

Comment:
You could even go further and specify `typing.Dict[str, str] = None`, but this would also require to change the test on `metadata={'foo': {'bar': 'baz'}}` in the same way you did for `metadata='foo'`